framework/include/bcs/LagrangeVecFunctionDirichletBC.h

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef LAGRANGEVECFUNCTIONDIRICHLETBC_H
+#define LAGRANGEVECFUNCTIONDIRICHLETBC_H
+
+#include "VectorNodalBC.h"
+
+class LagrangeVecFunctionDirichletBC;
+
+template <>
+InputParameters validParams<LagrangeVecFunctionDirichletBC>();
+
+/**
+ * Boundary condition of a Dirichlet type
+ *
+ * Sets the values of a LAGRANGE_VEC variable at nodes to values specified by functions
+ */
+class LagrangeVecFunctionDirichletBC : public VectorNodalBC
+{
+public:
+  LagrangeVecFunctionDirichletBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual(size_t comp) override;
```

Comment:
I'd prefer to use the C++ version for our new APIs. Let's not mix C and C++:
https://stackoverflow.com/questions/5813700/difference-between-size-t-and-stdsize-t

This is going to apply all over the place in this PR... Take a look over everything and and update appropriately.

Reply:
I know this seems like a big pain but note that according to the standard. You are only guaranteed to get the non-namespaced version if you include `stddef.h`. In C++, you would include `cstddef` and get only the namespaced version. Now in practice, we are going to always get both because real headers are almost never 100% clean. I think it's worth doing right.